scrapy/spiders/crawl.py

```diff
@@ -72,7 +73,7 @@ def _process_request(self, request, response):
 
 class CrawlSpider(Spider):
 
-    rules = ()
+    rules: Union[List[Rule], List, Tuple[Rule], Tuple] = ()
```

Comment:
I don't like having `List/Tuple` here, but `Incompatible types in assignment (expression has type "Tuple[]", variable has type "Union[List[Rule], Tuple[Rule]]")` occurs otherwise. I think it would be much more reasonable for `mypy` to consider an empty tuplr as a valid object.
An alternative would be `rules: Union[List[Rule], Tuple[Rule]] = ()  # type: ignore`, but that would mean the type hint is not actually being evaluated, it'd be there just for documentation purposes. Any thoughts?

See also https://www.python.org/dev/peps/pep-0484/#type-comments

Reply:
From what I’m reading, maybe `List[Rule, ...]` could replace `List[Rule], List`.

Reply:
I wonder if we should use `Sequence[Rule, ...]` here.

Reply:
Awesome, `rules: Sequence[Rule] = ()` works, many thanks!